packages/clerk-js/src/ui/components/PaymentSources/AddPaymentSource.tsx

```diff
@@ -0,0 +1,258 @@
+    } catch (error) {
+      handleError(error, [], setSubmitError);
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  return (
+    <Form
+      onSubmit={onSubmit}
+      sx={t => ({
+        display: 'flex',
+        flexDirection: 'column',
+        rowGap: t.space.$3,
+      })}
+    >
+      {collapsed ? (
+        <>
+          <Button
+            elementId={descriptors.button.setId('applePay')}
+            variant='unstyled'
+            size='md'
+            textVariant={'buttonLarge'}
+            sx={{
+              width: '100%',
+              backgroundColor: 'black',
+              color: 'white',
+            }}
+          >
+            {/* TODO(@COMMERCE): needs localization */}
+            Pay with ApplePay
+          </Button>
+          <Button
+            elementId={descriptors.button.setId('gPay')}
+            variant='unstyled'
+            size='md'
+            textVariant={'buttonLarge'}
+            block
+            sx={{
+              backgroundColor: 'black',
+              color: 'white',
+            }}
+          >
+            {/* TODO(@COMMERCE): needs localization */}
+            Pay with GPay
+          </Button>
+          <Button
+            colorScheme='secondary'
+            variant='bordered'
+            size='md'
+            textVariant={'buttonLarge'}
+            block
+            onClick={() => setCollapsed(false)}
+          >
+            {/* TODO(@COMMERCE): needs localization */}
+            More Payment Methods
+          </Button>
+          {cancelAction ? (
+            <Button
+              variant='ghost'
+              size='md'
+              textVariant={'buttonLarge'}
+              block
+              onClick={cancelAction}
+            >
+              {cancelButtonText ?? 'Cancel'}
+            </Button>
+          ) : null}
+        </>
+      ) : (
+        <>
+          <PaymentElement />
+          {submitError && (
+            <Alert
+              variant='danger'
+              sx={t => ({
+                animation: `${animations.textInBig} ${t.transitionDuration.$slow}`,
+              })}
+            >
+              {typeof submitError === 'string' ? submitError : submitError.message}
+            </Alert>
+          )}
+          <Flex gap={2}>
+            {cancelAction && !isSubmitting ? (
+              <Button
+                variant='ghost'
+                size='md'
+                textVariant={'buttonLarge'}
+                block
+                onClick={cancelAction}
+              >
+                {cancelButtonText ?? 'Cancel'}
+              </Button>
+            ) : null}
+            <Button
+              type='submit'
+              colorScheme='primary'
+              size='sm'
+              textVariant={'buttonLarge'}
+              block
+              isLoading={isSubmitting}
+            >
+              {submitButtonText ?? 'Add Payment Method'}
+            </Button>
+          </Flex>
```

Comment:
We should include the heading and button layout/styling that we use for adding an email address. We have a FormButtons component that we could maybe use here?

https://github.com/clerk/javascript/blob/5bd3cbb0a19e0b13b812453014aca9fc3418d98a/packages/clerk-js/src/ui/elements/FormButtons.tsx#L15-L33

![Screenshot 2025-03-31 at 2 05 28 PM](https://github.com/user-attachments/assets/f7c06fdf-0af2-40c9-9b66-0fa75a1e3be7)